api/service.go

```diff
@@ -351,19 +364,26 @@ func (s *service) Listen(req *pb.ListenRequest, server pb.API_ListenServer) erro
 		select {
 		case <-server.Context().Done():
 			return nil
-		case _, ok := <-l.Channel():
+		case action, ok := <-l.Channel():
 			if !ok {
 				return nil
 			}
-			err := model.ReadTxn(func(txn *es.Txn) error {
-				var res string
-				if err := txn.FindByID(core.EntityID(req.EntityID), &res); err != nil {
-					return err
-				}
-				return server.Send(&pb.ListenReply{
-					Entity: res,
-				})
-			})
+			var replyAction pb.ListenReply_Action
+			switch action.Type {
+			case es.ActionCreate:
+				replyAction = pb.ListenReply_CREATE
+			case es.ActionDelete:
+				replyAction = pb.ListenReply_DELETE
+			case es.ActionSave:
+				replyAction = pb.ListenReply_SAVE
+			}
+			// TODO: do we want to send the entity data in case of create and save?
```

Comment:
We were previously sending the entity data with the event. Is this a good idea? If so, we could include it only for create and save events? Is this a bad idea since we don't know how much data it will be?

Reply:
I decided yes, send the data.